ui/packages/catalog-ui-search/src/main/webapp/react-component/container/table-export/table-export.tsx

```diff
@@ -80,80 +71,79 @@ function getExportCount({
   if (exportSize === 'custom') {
     return customExportCount
   }
-
   const result = selectionInterface.getCurrentQuery().get('result')
   return exportSize === 'all'
     ? getHits(result.get('status').toJSON())
     : result.get('results').length
 }
-
 function getSorts(selectionInterface: any) {
   return selectionInterface.getCurrentQuery().get('sorts')
 }
-
 function getWarning(exportCountInfo: ExportCountInfo): string {
   const exportCount = getExportCount(exportCountInfo)
-  if (
-    exportCountInfo.exportSize === 'custom' &&
-    exportCount > properties.exportResultLimit
-  ) {
-    return `You cannot export more than the administrator configured limit of ${
-      properties.exportResultLimit
-    }`
-  }
-
   const result = exportCountInfo.selectionInterface
     .getCurrentQuery()
     .get('result')
   const totalHits = getHits(result.get('status').toJSON())
-
+  const limitWarning = `You cannot export more than the administrator configured limit of ${
+    properties.exportResultLimit
+  }.`
   let warningMessage = ''
-  if (exportCount > totalHits) {
-    warningMessage = `You are trying to export ${exportCount} results but there ${
-      totalHits === 1 ? `is` : `are`
-    } only ${totalHits}.  Only ${totalHits} ${
-      totalHits === 1 ? `result` : `results`
-    } will be exported.`
+  if (exportCount > properties.exportResultLimit) {
+    if (exportCountInfo.exportSize === 'custom') {
+      return limitWarning
+    }
+    warningMessage =
+      limitWarning +
+      `  Only ${properties.exportResultLimit} ${
+        properties.exportResultLimit === 1 ? `result` : `results`
+      } will be exported.`
   }
-  if (totalHits > 100 && exportCount > 100) {
-    warningMessage += ` This may take a long time.`
+  if (exportCountInfo.exportSize === 'custom') {
+    if (exportCount > totalHits) {
+      warningMessage = `You are trying to export ${exportCount} results but there ${
```

Comment:
✏️ This template is a little hard to read with the ternaries. Consider removing the logic for 1 vs multiple results and use `result(s)` instead. Alternatively, define two messages for the two templates.

Reply:
Alternatively, you could use `react-intl` to handle this logic. 

https://github.com/yahoo/react-intl